core/src/test/java/org/infinispan/conflict/impl/ConflictManagerTest.java

```diff
@@ -74,61 +77,53 @@ public void testGetAllVersionsDuringStateTransfer() throws Exception {
       createCluster();
       getCache(2).put(key, value);
       splitCluster();
-
-      delayStateTransferCompletion();
       RehashListener listener = new RehashListener();
-      getCache(2).addListener(listener);
+      getCache(0).addListener(listener);
+      CountDownLatch latch = new CountDownLatch(1);
+      delayStateTransferCompletion(latch);
 
       Future<Map<Address, InternalCacheValue<Object>>> versionFuture = fork(() -> {
-         while (!listener.isRehashInProgress.get())
-            TestingUtil.sleepThread(100);
-
+         listener.latch.await();
          return getAllVersions(0, key);
       });
       fork(() -> partition(0).merge(partition(1)));
+      new Timer().schedule(new TimerTask() {
```

Comment:
I really don't like tests based on timers... isn't possible to do the following:
```
listener.latch.await()
check not versionFuture.isDone()
latch.counterDown()
versionMap = versionFuture.get(...)
```